libs/server-sent-events/src/error.cpp

```diff
@@ -0,0 +1,61 @@
+#include <launchdarkly/sse/error.hpp>
+
+#include <sstream>
+
+namespace launchdarkly::sse {
+namespace errors {
+
+std::ostream& operator<<(std::ostream& out, NoContent const&) {
+    out << "no content, will not attempt to reconnect (HTTP 204)";
+    return out;
+}
+
+std::ostream& operator<<(std::ostream& out,
+                         InvalidRedirectLocation const& invalid) {
+    out << "server responded with an invalid redirect (" << invalid.location
+        << ")";
+    return out;
+}
+
+std::ostream& operator<<(std::ostream& out, NotRedirectable const&) {
+    out << "cannot follow server redirect";
+    return out;
+}
+
+std::ostream& operator<<(std::ostream& out, ReadTimeout const& err) {
+    out << "timed out reading response body (exceeded " << err.timeout->count()
+        << "ms)";
+    return out;
+}
+
+std::ostream& operator<<(std::ostream& out,
+                         UnrecoverableClientError const& err) {
+    if (err.status == boost::beast::http::status::unauthorized ||
+        err.status == boost::beast::http::status::forbidden) {
+        out << "invalid auth key (HTTP " << static_cast<int>(err.status) << ")";
```

Comment:
This could be more explicit (mobile vs SDK key), but that would require some special casing in each SDK and I don't think that logic is worth it. This is clear enough.